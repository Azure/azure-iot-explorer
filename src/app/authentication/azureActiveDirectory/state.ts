/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
export interface AzureActiveDirectoryStateInterface {
    formState: 'initialized' | 'working' | 'failed' | 'idle';
    token: string;
}

export const getInitialAzureActiveDirectoryState = (): AzureActiveDirectoryStateInterface => ({
    formState: 'initialized',
    token: undefined
});
