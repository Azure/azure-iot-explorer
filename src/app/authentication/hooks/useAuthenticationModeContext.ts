/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';

export enum AuthenticationMode {
    connectionString,
    roleBased
}

export const useAuthenticationMode = (): AuthenticationModeContextType => {
    const [ authenticationMode, setAuthenticationMode ] = React.useState<AuthenticationMode>(AuthenticationMode.connectionString);
    return {
        authenticationMode,
        setAuthenticationMode
    };
};

export interface AuthenticationModeContextType {
    authenticationMode: AuthenticationMode;
    setAuthenticationMode(authenticationMode: AuthenticationMode): void;
}

export const AuthenticationModeContext = React.createContext<AuthenticationModeContextType>({
    authenticationMode: AuthenticationMode.connectionString,
    setAuthenticationMode: () => {} // tslint:disable-line: no-empty
});
export const useAuthenticationModeContext = () => React.useContext<AuthenticationModeContextType>(AuthenticationModeContext);
