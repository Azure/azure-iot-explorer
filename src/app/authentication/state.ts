/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
export enum AuthenticationMethodPreference {
    AzureAD = 'AzureAD',
    ConnectionString = 'ConnectionString'
}

export interface AuthenticationStateInterface {
    formState: 'initialized' | 'working' | 'failed' | 'idle';
    preference: string;
}

export const getInitialAuthenticationState = (): AuthenticationStateInterface => ({
    formState: 'initialized',
    preference: undefined
});
