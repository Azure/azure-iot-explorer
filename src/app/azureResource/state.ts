/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { AzureResource } from './models/azureResource';

export interface AzureResourceStateInterface {
    activeAzureResource?: AzureResource;
}

export const azureResourceStateInitial = (): AzureResourceStateInterface => {
    return {};
};
