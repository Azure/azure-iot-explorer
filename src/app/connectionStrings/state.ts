/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { Record } from 'immutable';
import { IM } from '../shared/types/types';
import { SynchronizationWrapper } from '../api/models/synchronizationWrapper';
import { SynchronizationStatus } from '../api/models/synchronizationStatus';

export interface ConnectionStringsStateInterface extends SynchronizationWrapper<ConnectionStringWithExpiry[]>{}

export type ConnectionStringsStateType = IM<ConnectionStringsStateInterface>;

export interface ConnectionStringWithExpiry {
    connectionString: string;
    expiration: string;
}

export const connectionStringsStateInitial = Record<ConnectionStringsStateInterface>({
    payload: [],
    synchronizationStatus: SynchronizationStatus.initialized
});
