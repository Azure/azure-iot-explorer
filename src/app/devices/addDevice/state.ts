/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { Record } from 'immutable';
import { IM } from '../../shared/types/types';
import { SynchronizationStatus } from '../../api/models/synchronizationStatus';

export interface AddDeviceStateInterface {
    synchronizationStatus: SynchronizationStatus;
}

export const addDeviceStateInitial = Record<AddDeviceStateInterface>({
    synchronizationStatus: SynchronizationStatus.initialized
});

export type AddDeviceStateType = IM<AddDeviceStateInterface>;
