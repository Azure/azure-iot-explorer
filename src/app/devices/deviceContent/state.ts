/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { Record } from 'immutable';
import { IM } from '../../shared/types/types';
import { ModelDefinitionWithSourceWrapper } from '../../api/models/modelDefinitionWithSourceWrapper';
import { DeviceIdentityWrapper } from '../../api/models/deviceIdentityWrapper';
import { DeviceTwinWrapper } from './../../api/models/deviceTwinWrapper';
import { DigitalTwinInterfacePropertiesWrapper } from '../../api/models/digitalTwinInterfacePropertiesWrapper';

export interface DeviceContentStateInterface {
    deviceIdentity: DeviceIdentityWrapper;
    deviceTwin: DeviceTwinWrapper;
    digitalTwinInterfaceProperties: DigitalTwinInterfacePropertiesWrapper;
    interfaceIdSelected: string;
    modelDefinitionWithSource: ModelDefinitionWithSourceWrapper;
}

export const deviceContentStateInitial = Record<DeviceContentStateInterface>({
    deviceIdentity: null,
    deviceTwin: null,
    digitalTwinInterfaceProperties: null,
    interfaceIdSelected: '',
    modelDefinitionWithSource: null,
});

export type DeviceContentStateType = IM<DeviceContentStateInterface>;
