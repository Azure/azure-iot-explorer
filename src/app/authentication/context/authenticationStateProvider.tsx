/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { getInitialAuthenticationState } from '../state';
import { authenticationReducer } from '../reducers';
import { useAsyncSagaReducer } from '../../shared/hooks/useAsyncSagaReducer';
import { AuthenticationStateContext } from './authenticationStateContext';
import { getLoginPreferenceAction, setLoginPreferenceAction } from '../actions';
import { authenticationSaga } from '../saga';

export interface AuthenticationInterface {
    getLoginPreference(): void;
    setLoginPreference(preference: string): void;
}

export const AuthenticationStateContextProvider: React.FC = props => {
    const [state, dispatch] = useAsyncSagaReducer(authenticationReducer, authenticationSaga, getInitialAuthenticationState(), 'authenticationState');

    const authenticationApi: AuthenticationInterface = {
        getLoginPreference: () => dispatch(getLoginPreferenceAction.started()),
        setLoginPreference: (reference: string) =>  dispatch(setLoginPreferenceAction.started(reference))
    };

    return (
        <AuthenticationStateContext.Provider value={[state, authenticationApi]}>
            {props.children}
        </AuthenticationStateContext.Provider>
    );
};
