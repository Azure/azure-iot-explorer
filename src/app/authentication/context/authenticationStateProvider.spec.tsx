import * as React from 'react';
import { shallow } from 'enzyme';
import { AuthenticationStateContextProvider } from './authenticationStateProvider';
import { getInitialAuthenticationState } from '../state';
import * as AsyncSagaReducer from '../../shared/hooks/useAsyncSagaReducer';

describe('AuthenticationStateContextProvider', ()=> {
    jest.spyOn(AsyncSagaReducer, 'useAsyncSagaReducer').mockReturnValue([getInitialAuthenticationState(), jest.fn()]);

    it('matches snapshot', () => {
        const component = <AuthenticationStateContextProvider>
            <span>test</span>
        </AuthenticationStateContextProvider>;
        expect(shallow(component)).toMatchSnapshot();
    });
});
