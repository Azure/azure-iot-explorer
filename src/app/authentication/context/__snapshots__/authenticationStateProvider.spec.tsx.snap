// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationStateContextProvider matches snapshot 1`] = `
<ContextProvider
  value={
    Array [
      Object {
        "formState": "initialized",
        "preference": undefined,
      },
      Object {
        "getLoginPreference": [Function],
        "setLoginPreference": [Function],
      },
    ]
  }
>
  <span>
    test
  </span>
</ContextProvider>
`;
